### ISSUE TITLE
Cannot start docker engine in Hyper-V mode

### ISSUE THREAD

**Issue by mcDandy** [docker/for-win#11963]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: A513EA4B-D544-47D1-B5D5-C45A8516E272/20210822124642

### Actual behavior
After docker engine failed to load it crashed after clicking restart docker.

### Expected behavior
After switching from wsl2 to hyper-v I expected docker to run normally (WSL2 does not work again)

### Information

  - Windows Version: Windows 11 pro (22000.160)
  - Docker Desktop Version: 3.6.0
  - WSL2 or Hyper-V backend?  Hyper-V
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No
  This problem began with WSL stopping working after windows update and me switching to Hyper-V

### Steps to reproduce the behavior

  1. Switch docker to Hyper-V engine
  2. Wait untill engine fails (when crashes which crash handler does not catch - click continue)
  3. Attempt to restart using notification icon (cannot recreate, now always closes normally)

After attempting to recreate the bug it shows "Cannot enable Hyper-V service error", sometimes crashes (does not get cought by crash handler. Probably only when docker window closed to system tray) and closes (Hyper-V is enabled on windows)

<details>
  <summary>Crash details</summary>
  
  ```
Informace o vyvolání ladění JIT najdete na konci této zprávy,
nikoli v tomto dialogovém okně.

************** Text výjimky **************
Docker.Core.Backend.BackendQuitException: Cannot enable Hyper-V service
   v Docker.Engines.LinuxHyperVEngine.<DoCheckPreconditionsAsync>d__20.MoveNext() v C:\workspaces\master-merge\src\github.com\docker\pinata\win\src\Docker.Engines\LinuxHyperVEngine.cs:řádek 143
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() v C:\workspaces\master-merge\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:řádek 29
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() v C:\workspaces\master-merge\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:řádek 62
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Docker.ApiServices.StateMachines.EngineStateMachine.<StartAsync>d__14.MoveNext() v C:\workspaces\master-merge\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\EngineStateMachine.cs:řádek 69
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Docker.Engines.Engines.<StartAsync>d__21.MoveNext() v C:\workspaces\master-merge\src\github.com\docker\pinata\win\src\Docker.Engines\Engines.cs:řádek 114
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Docker.Core.TaskExtension.<Forget>d__0.MoveNext() v C:\workspaces\master-merge\src\github.com\docker\pinata\win\src\Docker.Core\Extensions.cs:řádek 30
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()


************** Načtená sestavení **************
mscorlib
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4400.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
Docker Desktop
    Verze sestavení: 1.0.0.0
    Verze Win32: 3.6.0.67351
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Docker%20Desktop.exe
----------------------------------------
Docker.Core
    Verze sestavení: 1.0.0.0
    Verze Win32: 3.6.0.67351
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Docker.Core.DLL
----------------------------------------
Docker.WPF
    Verze sestavení: 1.0.0.0
    Verze Win32: 3.6.0.67351
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Docker.WPF.DLL
----------------------------------------
PresentationFramework
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.dll
----------------------------------------
WindowsBase
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0 built by: NET48REL1LAST_B
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
----------------------------------------
System.Core
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0 built by: NET48REL1LAST_B
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4360.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
PresentationCore
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0 built by: NET48REL1LAST_B
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/PresentationCore/v4.0_4.0.0.0__31bf3856ad364e35/PresentationCore.dll
----------------------------------------
System.Xaml
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0 built by: NET48REL1LAST_B
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xaml/v4.0_4.0.0.0__b77a5c561934e089/System.Xaml.dll
----------------------------------------
Docker.ApiServices
    Verze sestavení: 1.0.0.0
    Verze Win32: 3.6.0.67351
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Docker.ApiServices.DLL
----------------------------------------
Docker.Engines
    Verze sestavení: 1.0.0.0
    Verze Win32: 3.6.0.67351
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Docker.Engines.DLL
----------------------------------------
Docker.HttpApi
    Verze sestavení: 1.0.0.0
    Verze Win32: 3.6.0.67351
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Docker.HttpApi.DLL
----------------------------------------
System.Windows.Forms
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4400.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0 built by: NET48REL1LAST_B
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Net.Http
    Verze sestavení: 4.2.0.0
    Verze Win32: 4.6.26011.01
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Net.Http.DLL
----------------------------------------
HttpOverStream.NamedPipe
    Verze sestavení: 1.0.0.0
    Verze Win32: 1.0.0.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/HttpOverStream.NamedPipe.DLL
----------------------------------------
NLog
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.5.10.8381
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/NLog.DLL
----------------------------------------
System.Configuration
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Newtonsoft.Json
    Verze sestavení: 11.0.0.0
    Verze Win32: 11.0.2.21924
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Newtonsoft.Json.DLL
----------------------------------------
System.Numerics
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Runtime.Serialization
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.Data
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.ServiceModel
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ServiceModel/v4.0_4.0.0.0__b77a5c561934e089/System.ServiceModel.dll
----------------------------------------
BITSReference5_0
    Verze sestavení: 1.0.0.0
    Verze Win32: 1.0.0.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/BITSReference5_0.DLL
----------------------------------------
HttpOverStream
    Verze sestavení: 1.0.0.0
    Verze Win32: 1.0.0.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/HttpOverStream.DLL
----------------------------------------
System.Threading.Tasks.Extensions
    Verze sestavení: 4.2.0.1
    Verze Win32: 4.6.27818.01
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Threading.Tasks.Extensions.DLL
----------------------------------------
System.Web.Http
    Verze sestavení: 5.2.7.0
    Verze Win32: 5.2.61128.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Web.Http.DLL
----------------------------------------
HttpOverStream.Client
    Verze sestavení: 1.0.0.0
    Verze Win32: 1.0.0.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/HttpOverStream.Client.DLL
----------------------------------------
netstandard
    Verze sestavení: 2.0.0.0
    Verze Win32: 4.8.4161.0
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/netstandard/v4.0_2.0.0.0__cc7b13ffcd2ddd51/netstandard.dll
----------------------------------------
System.ValueTuple
    Verze sestavení: 4.0.3.0
    Verze Win32: 4.6.26515.06
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.ValueTuple.DLL
----------------------------------------
Bugsnag
    Verze sestavení: 2.2.0.0
    Verze Win32: 2.2.0.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Bugsnag.DLL
----------------------------------------
PresentationFramework.Aero2
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework.Aero2/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.Aero2.dll
----------------------------------------
PresentationFramework.resources
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework.resources/v4.0_4.0.0.0_cs_31bf3856ad364e35/PresentationFramework.resources.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Verze sestavení: 0.0.0.0
    Verze Win32: 4.8.4400.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
----------------------------------------
Microsoft.Toolkit.Uwp.Notifications
    Verze sestavení: 1.5.1.0
    Verze Win32: 1.5.1.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Microsoft.Toolkit.Uwp.Notifications.DLL
----------------------------------------
System.Runtime
    Verze sestavení: 4.1.2.0
    Verze Win32: 4.6.25714.01
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Runtime.DLL
----------------------------------------
Windows.UI
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.UI.winmd
----------------------------------------
Windows.Foundation
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Foundation.winmd
----------------------------------------
System.Runtime.InteropServices.WindowsRuntime
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.InteropServices.WindowsRuntime/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Runtime.InteropServices.WindowsRuntime.dll
----------------------------------------
Windows.Data
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Data.winmd
----------------------------------------
System.Collections
    Verze sestavení: 4.0.11.0
    Verze Win32: 4.6.25714.01
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Collections.DLL
----------------------------------------
System.Runtime.CompilerServices.Unsafe
    Verze sestavení: 4.0.4.1
    Verze Win32: 4.6.26919.02
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Runtime.CompilerServices.Unsafe.DLL
----------------------------------------
mscorlib.resources
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_cs_b77a5c561934e089/mscorlib.resources.dll
----------------------------------------
Microsoft.Owin
    Verze sestavení: 4.1.0.0
    Verze Win32: 4.1.81112.127
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Microsoft.Owin.DLL
----------------------------------------
Owin
    Verze sestavení: 1.0.0.0
    Verze Win32: 1.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Owin.DLL
----------------------------------------
HttpOverStream.Server.Owin
    Verze sestavení: 1.0.0.0
    Verze Win32: 1.0.0.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/HttpOverStream.Server.Owin.DLL
----------------------------------------
Microsoft.Owin.Hosting
    Verze sestavení: 4.1.0.0
    Verze Win32: 4.1.81112.127
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/Microsoft.Owin.Hosting.DLL
----------------------------------------
System.Web.Http.Owin
    Verze sestavení: 5.2.7.0
    Verze Win32: 5.2.61128.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Web.Http.Owin.DLL
----------------------------------------
System.Net.Http.Formatting
    Verze sestavení: 5.2.7.0
    Verze Win32: 5.2.61128.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Net.Http.Formatting.DLL
----------------------------------------
SMDiagnostics
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/SMDiagnostics/v4.0_4.0.0.0__b77a5c561934e089/SMDiagnostics.dll
----------------------------------------
System.ServiceProcess
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ServiceProcess/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.ServiceProcess.dll
----------------------------------------
UIAutomationProvider
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0 built by: NET48REL1LAST_B
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/UIAutomationProvider/v4.0_4.0.0.0__31bf3856ad364e35/UIAutomationProvider.dll
----------------------------------------
System.Security
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Security/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Security.dll
----------------------------------------
Accessibility
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
System.Data.SqlXml
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Data.SqlXml/v4.0_4.0.0.0__b77a5c561934e089/System.Data.SqlXml.dll
----------------------------------------
System.ServiceModel.Internals
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ServiceModel.Internals/v4.0_4.0.0.0__31bf3856ad364e35/System.ServiceModel.Internals.dll
----------------------------------------
System.Transactions
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Transactions/v4.0_4.0.0.0__b77a5c561934e089/System.Transactions.dll
----------------------------------------
System.IdentityModel
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4341.0 built by: NET48REL1LAST_C
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.IdentityModel/v4.0_4.0.0.0__b77a5c561934e089/System.IdentityModel.dll
----------------------------------------
Microsoft.Transactions.Bridge
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Microsoft.Transactions.Bridge/v4.0_4.0.0.0__b03f5f7f11d50a3a/Microsoft.Transactions.Bridge.dll
----------------------------------------
System.EnterpriseServices
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.EnterpriseServices/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.EnterpriseServices.dll
----------------------------------------
System.Messaging
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Messaging/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Messaging.dll
----------------------------------------
System.Runtime.DurableInstancing
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.DurableInstancing/v4.0_4.0.0.0__31bf3856ad364e35/System.Runtime.DurableInstancing.dll
----------------------------------------
System.Web.Services
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Web.Services/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Web.Services.dll
----------------------------------------
System.Runtime.WindowsRuntime
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.WindowsRuntime/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.WindowsRuntime.dll
----------------------------------------
System.Numerics.Vectors
    Verze sestavení: 4.1.4.0
    Verze Win32: 4.6.26515.06
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Numerics.Vectors.DLL
----------------------------------------
Windows.Graphics
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Graphics.winmd
----------------------------------------
Windows.System
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.System.winmd
----------------------------------------
Windows.Devices
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Devices.winmd
----------------------------------------
Windows.Gaming
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Gaming.winmd
----------------------------------------
Windows.ApplicationModel
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.ApplicationModel.winmd
----------------------------------------
Windows.Storage
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Storage.winmd
----------------------------------------
Windows.Web
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Web.winmd
----------------------------------------
Windows.Media
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Media.winmd
----------------------------------------
Windows.Security
    Verze sestavení: 255.255.255.255
    Verze Win32: 10.0.10011.16384
    Základ kódu (CodeBase): file:///C:/WINDOWS/system32/WinMetadata/Windows.Security.winmd
----------------------------------------
System.Configuration.Install
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration.Install/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.Install.dll
----------------------------------------
System.Xml.Linq
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
System.ComponentModel.DataAnnotations
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ComponentModel.DataAnnotations/v4.0_4.0.0.0__31bf3856ad364e35/System.ComponentModel.DataAnnotations.dll
----------------------------------------
Microsoft.VisualBasic
    Verze sestavení: 10.0.0.0
    Verze Win32: 14.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualBasic.dll
----------------------------------------
System.Management
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Management/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Management.dll
----------------------------------------
System.Net.IPNetwork
    Verze sestavení: 2.1.2.0
    Verze Win32: 2.1.2.0
    Základ kódu (CodeBase): file:///C:/Program%20Files/Docker/Docker/System.Net.IPNetwork.DLL
----------------------------------------
System.resources
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.resources/v4.0_4.0.0.0_cs_b77a5c561934e089/System.resources.dll
----------------------------------------
PresentationCore.resources
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationCore.resources/v4.0_4.0.0.0_cs_31bf3856ad364e35/PresentationCore.resources.dll
----------------------------------------
PresentationFramework-SystemXmlLinq
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework-SystemXmlLinq/v4.0_4.0.0.0__b77a5c561934e089/PresentationFramework-SystemXmlLinq.dll
----------------------------------------
PresentationFramework-SystemXml
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework-SystemXml/v4.0_4.0.0.0__b77a5c561934e089/PresentationFramework-SystemXml.dll
----------------------------------------
UIAutomationTypes
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4395.0 built by: NET48REL1LAST_B
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/UIAutomationTypes/v4.0_4.0.0.0__31bf3856ad364e35/UIAutomationTypes.dll
----------------------------------------
System.Windows.Forms.resources
    Verze sestavení: 4.0.0.0
    Verze Win32: 4.8.4161.0 built by: NET48REL1
    Základ kódu (CodeBase): file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_cs_b77a5c561934e089/System.Windows.Forms.resources.dll
----------------------------------------

************** Ladění JIT **************
Aby bylo povoleno ladění JIT, musí konfigurační soubor
pro tuto aplikaci nebo počítač (machine.config) mít
v oddílu system.windows.forms nastavenou hodnotu njitDebugging.
Aplikace rovněž musí být kompilována s povoleným
laděním.

Příklad:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

Je-li ladění JIT povoleno, budou všechny neošetřené výjimky
odeslány ladicímu programu JIT, který je registrován v počítači,
a nebudou zpracovány tímto dialogovým oknem.



```
</details>

**Comment by kondelik** [comment]
Why is this issue in ["release notes: whats new"](https://docs.docker.com/desktop/windows/release-notes/ ) as fixed but there isnt a single word?

Is it fixed or not?

**Comment by nicholasgcotton** [comment]
Still happening after update and reboot. Docker hangs for an extended period before saying it cannot enable Hyper-V (while Hyper-V is in fact up and running just fine). Manually re-starting docker works fine the second time.

**Comment by vantruong1810** [comment]
Seemly, It was fixed from v4.0.1 https://docs.docker.com/desktop/windows/release-notes/

**Comment by edorgeville** [comment]
Open an administrator command prompt and run this:
`bcdedit /set hypervisorlaunchtype auto`
Then reboot.

This fixed the issue for me on Docker Desktop 4.0.0.

Source:
https://github.com/docker/for-win/issues/6772#issuecomment-684660619
